PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java

```diff
@@ -0,0 +1,280 @@
+        buttonCancel.addActionListener(e -> onCancel());
+
+        rdoLinuxOS.addActionListener(e -> selectOS(false));
+        rdoWindowsOS.addActionListener(e -> selectOS(false));
+
+        final ButtonGroup osButtonGroup = new ButtonGroup();
+        osButtonGroup.add(rdoLinuxOS);
+        osButtonGroup.add(rdoWindowsOS);
+
+
+        rootPane.registerKeyboardAction(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                onCancel();
+            }
+        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
+
+        // call onCancel() when cross is clicked
+        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
+        addWindowListener(new WindowAdapter() {
+            public void windowClosing(WindowEvent e) {
+                onCancel();
+            }
+        });
+
+        addValidationListener(contentPanel, e -> validateConfiguration());
+        init();
+
+        subscriptionPanel.addItemListener(e -> {
+            final String subscriptionId = subscriptionPanel.getSubscriptionId();
+            if (subscriptionId != null) {
+                resourceGroupPanel.loadResourceGroup(subscriptionId);
+                appServicePlanPanel.loadAppServicePlan(subscriptionId, getSelectedOperationSystemEnum());
+            }
+        });
+    }
+
+    public FunctionApp getCreatedWebApp() {
+        return this.result;
+    }
+
+    public OperatingSystem getSelectedOperationSystemEnum() {
+        return rdoWindowsOS.isSelected() ? OperatingSystem.WINDOWS : OperatingSystem.LINUX;
+    }
+
+    private void addValidationListener(Container parent, ActionListener actionListener) {
+        for (final Component component : parent.getComponents()) {
+            if (component instanceof AbstractButton) {
+                ((AbstractButton) component).addActionListener(actionListener);
+            } else if (component instanceof JComboBox) {
+                ((JComboBox) component).addActionListener(actionListener);
+            } else if (component instanceof JTextField) {
+                ((JTextField) component).getDocument().addDocumentListener(new DocumentListener() {
+                    @Override
+                    public void insertUpdate(DocumentEvent e) {
+                        validateConfiguration();
+                    }
+
+                    @Override
+                    public void removeUpdate(DocumentEvent e) {
+                        validateConfiguration();
+                    }
+
+                    @Override
+                    public void changedUpdate(DocumentEvent e) {
+                        validateConfiguration();
+                    }
+                });
+            } else if (component instanceof Container) {
+                addValidationListener((Container) component, actionListener);
+            }
+        }
+    }
+
+    private void selectOS(boolean isWindows) {
```

Comment:
param isWindows is not used?

Reply:
This parameter has been out of date and should be removed